src/components/tab-bar/tab-bar.tsx

```diff
@@ -0,0 +1,142 @@
+const { TAB_ACTIVATED_EVENT } = strings;
+
+@Component({
+    tag: 'limel-tab-bar',
+    styleUrl: 'tab-bar.scss',
+    shadow: true,
+})
+export class TabBar {
+    @Element()
+    private element: HTMLElement;
+
+    /**
+     * List of tabs to display
+     */
+    @Prop()
+    public tabs: Tab[];
+
+    private mdcTabBar: MDCTabBar;
+
+    @Event()
+    private changeTab: EventEmitter<Tab>;
+
+    @State()
+    private selectedTab = this.tabs[0];
+
+    constructor() {
+        this.handleTabActivated = this.handleTabActivated.bind(this);
+        this.renderTab = this.renderTab.bind(this);
+    }
+
+    public componentDidLoad() {
+        this.setup();
+    }
+
+    private setup() {
+        const element = this.element.shadowRoot.querySelector('.mdc-tab-bar');
+        if (!element) {
+            return;
+        }
+
+        this.mdcTabBar = new MDCTabBar(element);
+
+        // Workaround for shadow dom support for material
+        (this
+            .mdcTabBar as any).foundation_.adapter_.getFocusedTabIndex = () => {
+            const tabElements = this.getTabElements();
+            const activeElement = this.element.shadowRoot.activeElement;
+            return tabElements.indexOf(activeElement);
+        };
+
+        this.setupListeners();
+    }
+
+    private getTabElements() {
+        return [].slice.call(
+            this.element.shadowRoot.querySelectorAll('.mdc-tab')
+        );
+    }
+
+    private setupListeners() {
+        this.mdcTabBar.listen(TAB_ACTIVATED_EVENT, this.handleTabActivated);
+    }
+
+    private handleTabActivated(event: MDCTabBarActivatedEvent) {
+        const index = event.detail.index;
+        const selectedTab = this.tabs[index];
+        this.selectedTab = selectedTab;
+        this.changeTab.emit({ ...this.selectedTab, active: true });
+    }
+
+    render() {
+        return (
+            <div class="mdc-tab-bar" role="tablist">
+                <div class="mdc-tab-scroller">
+                    <div class="mdc-tab-scroller__scroll-area mdc-tab-scroller__scroll-area--scroll">
+                        <div class="mdc-tab-scroller__scroll-content">
+                            {this.tabs.map(this.renderTab)}
+                        </div>
+                    </div>
+                </div>
+            </div>
+        );
+    }
+
+    private renderIcon(tab: Tab) {
+        const style = {};
+        if (tab.iconColor) {
+            style['background-color'] = tab.iconColor;
+            style.color = 'white';
+        }
+
+        return (
+            <limel-icon
+                class="mdc-tab-bar__icon"
+                name={tab.icon}
+                badge={true}
+                style={style}
+                size="small"
+            />
+        );
+    }
+
+    private renderTab(tab: Tab) {
+        return (
+            <button
+                class={`
+                mdc-tab
+                ${tab.active ? 'mdc-tab--active' : ''}
+            `}
```

Comment:
I think inline string literals for attributes makes the code a bit messy hand hard to read. I suggest we declare them as an object at the top of the function instead, e.g.
```tsx
private renderTab(tab: Tab) {
    const classList = {
        'mdc-tab': true,
        'mdc-tab--active': !!tab.active
    };

    return (
        <button class={classList}>...</button>
    );
}
```